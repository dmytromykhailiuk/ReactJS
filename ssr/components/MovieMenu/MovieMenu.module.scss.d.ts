// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'movie-menu': string;
  'movie-menu__button': string;
}
export const cssExports: CssExports;
export default cssExports;
