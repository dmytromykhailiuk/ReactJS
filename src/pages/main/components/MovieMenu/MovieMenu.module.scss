@import "scss/variables";
@import "scss/mixins";

.movie-menu {
  background-color: $dark-grey-color;
  box-shadow: 0 2px 6px rgb(0, 0, 0);
  width: 50vw;
  border-radius: 1vw;
  padding: 1vw 0;

  @include small-screen {
    width: 25vw;
    border-radius: 0.5vw;
    padding: 0.5vw 0;
  }

  @include large-screen {
    width: 16.6vw;
    border-radius: 0.33vw;
    padding: 0.33vw 0;
  }

  @include extra-large-screen {
    width: 12.6vw;
    border-radius: 0.25vw;
    padding: 0.25vw 0;
  }

  &__button {
    color: $white-color;
    font-size: 3.5vw;
    padding: 2vw 6vw;
    letter-spacing: 0.15vw;
    transition: 0.4s;

    @include small-screen {
      font-size: 1.75vw;
      padding: 1vw 3vw;
      letter-spacing: 0.075vw;
    }

    @include large-screen {
      font-size: 1.16vw;
      padding: 0.66vw 2vw;
      letter-spacing: 0.05vw;
    }

    @include extra-large-screen {
      font-size: 0.88vw;
      padding: 0.5vw 1.52vw;
      letter-spacing: 0.038vw;
    }

    &:hover {
      background-color: $pink-color;
    }
  }
}
