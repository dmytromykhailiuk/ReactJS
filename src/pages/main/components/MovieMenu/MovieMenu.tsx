import React, { useEffect, useMemo } from 'react';
import classes from './MovieMenu.module.scss';
import CloseMenuButton from 'shared/components/CloseMenuButton/CloseMenuButton';
import { clickOutside } from 'shared/helpers';

interface MovieMenuProps {
  uniqueClass: string;
  onCloseButtonClicked?: () => void;
  onEditButtonClicked?: () => void;
  onDeleteButtonClicked?: () => void;
}

const MovieMenu: React.FC<MovieMenuProps> = ({
  uniqueClass,
  onCloseButtonClicked = () => {},
  onEditButtonClicked = () => {},
  onDeleteButtonClicked = () => {},
}) => {
  const clickOutsides = useMemo(() => clickOutside(uniqueClass, onCloseButtonClicked), [])

  useEffect(() => {
    document.addEventListener("click", clickOutsides);
    return () => {
      document.removeEventListener("click", clickOutsides);
    }
  }, [])

  return (
    <div className={`${uniqueClass} ${classes["movie-menu"]}`}>
      <CloseMenuButton isSmall={true} onCloseButtonClicked={onCloseButtonClicked} />
      <div className={classes["movie-menu__button"]} onClick={onEditButtonClicked} >Edit</div>
      <div className={classes["movie-menu__button"]} onClick={onDeleteButtonClicked}>Delete</div>
    </div>
  )
}

export default MovieMenu;
