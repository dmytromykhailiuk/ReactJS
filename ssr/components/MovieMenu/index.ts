export { default as MovieMenu } from "./MovieMenu";
