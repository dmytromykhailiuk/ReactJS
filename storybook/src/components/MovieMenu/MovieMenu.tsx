import React from 'react';
import CloseMenuButton from '../CloseMenuButton/CloseMenuButton';
import { useClickOutside } from '../../hooks';
import classes from './MovieMenu.module.scss';

interface MovieMenuProps {
  uniqueClass: number;
  onCloseButtonClicked: () => void;
  onEditButtonClicked: () => void;
  onDeleteButtonClicked: () => void;
}

const MovieMenu: React.FC<MovieMenuProps> = ({
  uniqueClass,
  onCloseButtonClicked,
  onEditButtonClicked,
  onDeleteButtonClicked,
}) => {
  useClickOutside(String(uniqueClass), onCloseButtonClicked);

  return (
    <div className={`${uniqueClass} ${classes['movie-menu']}`}>
      <CloseMenuButton isSmall onCloseButtonClicked={onCloseButtonClicked} />
      <div className={classes['movie-menu__button']} onClick={onEditButtonClicked}>
        Edit
      </div>
      <div className={classes['movie-menu__button']} onClick={onDeleteButtonClicked}>
        Delete
      </div>
    </div>
  );
};

export default MovieMenu;
