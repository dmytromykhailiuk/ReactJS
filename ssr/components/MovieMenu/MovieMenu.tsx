import React from 'react';
import classes from './MovieMenu.module.scss';
import { CloseMenuButton } from '../../shared/components';
import { useClickOutside } from '../../shared/hooks';

interface MovieMenuProps {
  uniqueClass: number;
  onCloseButtonClicked: () => void;
  onEditButtonClicked: () => void;
  onDeleteButtonClicked: () => void;
}

const MovieMenu: React.FC<MovieMenuProps> = ({
  uniqueClass,
  onCloseButtonClicked,
  onEditButtonClicked,
  onDeleteButtonClicked,
}) => {
  useClickOutside(String(uniqueClass), onCloseButtonClicked);

  return (
    <div className={`${uniqueClass} ${classes['movie-menu']}`}>
      <CloseMenuButton isSmall={true} onCloseButtonClicked={onCloseButtonClicked} />
      <div className={classes['movie-menu__button']} onClick={onEditButtonClicked}>
        Edit
      </div>
      <div className={classes['movie-menu__button']} onClick={onDeleteButtonClicked}>
        Delete
      </div>
    </div>
  );
};

export default MovieMenu;
