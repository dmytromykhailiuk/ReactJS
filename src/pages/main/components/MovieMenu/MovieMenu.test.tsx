import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import MovieMenu from './MovieMenu';

describe('MovieMenu', () => {
  it('should match snepshot', () => {
    const wrapper = shallow(
      <MovieMenu
        uniqueClass={12111}
        onCloseButtonClicked={() => {}}
        onEditButtonClicked={() => {}}
        onDeleteButtonClicked={() => {}}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
