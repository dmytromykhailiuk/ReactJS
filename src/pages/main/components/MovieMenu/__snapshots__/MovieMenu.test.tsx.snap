// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieMenu should match snepshot 1`] = `
<div
  className="12111 movie-menu"
>
  <CloseMenuButton
    isSmall={true}
    onCloseButtonClicked={[Function]}
  />
  <div
    className="movie-menu__button"
    onClick={[Function]}
  >
    Edit
  </div>
  <div
    className="movie-menu__button"
    onClick={[Function]}
  >
    Delete
  </div>
</div>
`;
